### PR TITLE
New salad form

### PR DESCRIPTION
**Pull request by bobbydigital1986**
Requires a salad name
Description is optional
Stays on the page and returns errors if salad name not entered
Redirects user to salad’s show page after successful submission
Hide new salad button from navbar if not logged in
Redirect to registration page if non-logged in user navaigates to /salads/new

### CODE REVIEW

**Review by ToddLGarrison — COMMENTED** [review]
Looks good to me

File: client/src/components/layout/TopBar.js
Comment (Jesspinoza) [inline client/src/components/layout/TopBar.js:1]: Looks good!

**Review by Jesspinoza — COMMENTED** [review]
this looks good

File: client/src/components/App.js [hunk 1]
```diff
@@ -26,11 +27,19 @@ const App = (props) => {
...
+    newPostComponent = RegistrationForm
+  }
+
   return (
     <Router>
       <TopBar user={currentUser} />
       <Switch>
-        <Route exact path="/salads" component={SaladListComponent} />
+        <Route exact path="/salads/new" component={newPostComponent}></Route>
+        <Route exact path="/salads" component={SaladListComponent}></Route>
```
Comment (nwalberts) [inline client/src/components/App.js:42]: neither line 41 or 42 need a closing </Route> tag. Just make the initial one self-closing like the others

File: client/src/components/App.js [hunk 2]
```diff
@@ -26,11 +27,19 @@ const App = (props) => {
     fetchCurrentUser()
   }, [])
 
+  let newPostComponent
+  if (currentUser) {
+    newPostComponent = NewSaladForm
+  } else {
+    newPostComponent = RegistrationForm
+  }
+
```
Comment (nwalberts) [inline client/src/components/App.js:36]: nope, unfortunately Im not going to allow this. You need to use the `AuthenticatedRoute` pattern shown in the user auth clinic and readings

File: client/src/components/layout/TopBar.js [hunk 3]
```diff
@@ -20,6 +20,15 @@ const TopBar = ({ user }) => {
     </li>,
   ];
 
+  let newPostLink
+  if (user) {
+    newPostLink = (
+      <li className="menu-text">
+        <Link to="/salads/new">Post a salad!</Link>
+      </li>
```
Comment (nwalberts) [inline client/src/components/layout/TopBar.js:28]: this doesnt need to be an li. delete the li. If you need the styling, then you can add a div, or apply the styling as a className directly to the Link

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Review by nwalberts — APPROVED** [review]

**Review by nwalberts — APPROVED** [review]
good

**Pull request merged**